src/main/java/duke/Task.java

```diff
@@ -0,0 +1,36 @@
+package duke;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description, boolean isDone) throws DukeException {
+
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmarks the task as not done.
+     */
+    public void unmark() {
+        this.isDone = false;
+    }
```

Comment:
Perhaps you could consider naming these two methods similarly since they perform similar functions? 

i.e. Renaming unmark to markAsUndone, or markAsDone to simply mark

Reply:
Don't think this is possible because I have 2 commands, mark and unmark. If I combine the two methods I won't be able to check whether it's done or not done.